lib/src/map/flutter_map_state.dart

```diff
@@ -516,19 +515,21 @@ class FlutterMapState extends MapGestureMixin
 
   Offset getOffsetFromOrigin(LatLng pos) {
     final delta = project(pos) - _pixelOrigin;
-    return Offset(delta.x.toDouble(), delta.y.toDouble());
+    return Offset(delta.x, delta.y);
   }
 
   CustomPoint<int> getNewPixelOrigin(LatLng center, [double? zoom]) {
-    final viewHalf = size / 2.0;
-    return (project(center, zoom) - viewHalf).round();
+    final halfSize = size / 2.0;
+    return (project(center, zoom) - halfSize).round();
   }
 
-  Bounds<double> getPixelBounds(double zoom) {
-    final mapZoom = zoom;
```

Comment:
I believe this to be a bug. It should have been:

    final mapZoom = this.zoom;

Otherwise scale will always be `1.0` :shrug:

Reply:
Interesting spot! Will have to test this just to make sure there's no side effects.